packages/indexer-common/src/indexer-management/subgraphs.ts

```diff
@@ -0,0 +1,200 @@
+      })
+      const requestPromise = this.client.request('subgraph_deploy', {
+        name,
+        ipfs_hash: deployment.ipfsHash,
+        node_id: targetNode,
+      })
+      // Timeout deployment after 2 minutes
+      const response = await pTimeout(requestPromise, 120000)
+
+      if (response.error) {
+        throw response.error
+      }
+      logger.info(`Successfully deployed subgraph`, {
+        name,
+        deployment: deployment.display,
+        response,
+      })
+
+      logger.info(
+        `Update indexing rules, so indexer-agent keeps the deployment synced but doesn't allocate to it`,
+      )
+      const offchainIndexingRule = {
+        identifier: deployment.ipfsHash,
+        identifierType: SubgraphIdentifierType.DEPLOYMENT,
+        decisionBasis: IndexingDecisionBasis.OFFCHAIN,
+      } as Partial<IndexingRuleAttributes>
+
+      await models.IndexingRule.upsert(offchainIndexingRule)
+
+      // Since upsert succeeded, we _must_ have a rule
+      const updatedRule = await models.IndexingRule.findOne({
+        where: { identifier: offchainIndexingRule.identifier },
+      })
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      logger.info(`DecisionBasis.OFFCHAIN rule merged into indexing rules`, {
+        rule: updatedRule,
+      })
+    } catch (error) {
+      const err = indexerError(IndexerErrorCode.IE026, error)
+      logger.error(`Failed to deploy subgraph deployment`, {
+        name,
+        deployment: deployment.display,
+        err,
+      })
+      throw err
+    }
+  }
+
+  async remove(
+    logger: Logger,
+    models: IndexerManagementModels,
+    deployment: SubgraphDeploymentID,
+  ): Promise<void> {
+    try {
+      logger.info(`Remove subgraph deployment`, {
+        deployment: deployment.display,
+      })
+      const response = await this.client.request('subgraph_reassign', {
+        node_id: 'removed',
+        ipfs_hash: deployment.ipfsHash,
+      })
+      if (response.error) {
+        throw response.error
+      }
+      logger.info(`Successfully removed subgraph deployment`, {
+        deployment: deployment.display,
+      })
+
+      if (
+        await models.IndexingRule.findOne({
+          where: { identifier: deployment.ipfsHash },
+        })
+      ) {
+        logger.info(
+          `Remove indexing rules, so indexer-agent will not attempt to redeploy`,
+        )
+        await models.IndexingRule.destroy({
+          where: {
+            identifier: deployment.ipfsHash,
+          },
+        })
+        logger.info(`Sucessfully removed indexing rule for '${deployment.ipfsHash}'`)
+      }
+    } catch (error) {
+      const err = indexerError(IndexerErrorCode.IE027, error)
+      logger.error(`Failed to remove subgraph deployment`, {
+        deployment: deployment.display,
+        err,
+      })
+    }
+  }
+
+  async reassign(logger: Logger, deployment: SubgraphDeploymentID): Promise<void> {
```

Comment:
Maybe it's worth exposing this as a GraphQL mutation? Obviously one can call the `reassign` rpc directly on a a graph-node, but it might be nice to expose this to make indexer-agent's API comprehensive and avoid calling dependencies directly.

Reply:
Absolutely.  I actually have those resolvers written up and stashed; I decided to table that change for a subsequent PR since they aren't essential to allocation management and I'd like to include a corresponding indexer-cli module to go with them (`indexer subgraphs deploy`, etc..).